### PR TITLE
Search Bar

### PR DESCRIPTION
**Pull request by thaitwo**
When you have some time, can you please take a look over? Thanks!! 🍵 ☕️ 🍏 

- Added if statement in search-bar.jsx to prevent the URL from updating to "example.com/weather///"when users click on submit button with an empty field
- Added a few css classes to hide the search bar suggestions when users click outside of it
- Added componentDidUpdate() to allow new Ajax requests when users submit a new city

### CODE REVIEW

File: dev/jsx/search-bar.jsx [hunk 1]
```diff
@@ -20,7 +20,9 @@ class SearchBar extends React.Component {
   // When user submits a city, create a link using the data obtained from the selected suggestion
   handleSubmit(event) {
     event.preventDefault();
-    browserHistory.push(`/weather/${this.state.country}/${this.state.city}/${this.state.state}`);
+    if (this.state.city && this.state.state && this.state.country) {
```
Comment (alexwolfe) [inline dev/jsx/search-bar.jsx:23]: 👍  Good check!

File: dev/jsx/weather.jsx [hunk 2]
```diff
@@ -48,8 +48,23 @@ class FetchWeather extends React.Component {
...
+  }
+
+  // Compare the previous props with the current props taken from the URL.
+  // If they are different, then make a new Ajax request to get new data.
+  // This comparison prevents recursive Ajax requests.
+  componentDidUpdate(prevProps) {
+    const {city, state, country} = this.props.params;
+    const prevCity = prevProps.params.city;
+    const prevState = prevProps.params.state;
+    const prevCountry = prevProps.params.country;
```
Comment (alexwolfe) [inline dev/jsx/weather.jsx]: You can do the same thing here to be more efficient when declaring vars: 
```
const {prevCity, prevState, prevCountry} = prevProps.params;
```

File: dev/scss/_search-bar.scss [hunk 3]
```diff
@@ -79,4 +94,13 @@ form {
   &:hover {
     color: rgba(0, 0, 0, .3);
   }
-}
\ No newline at end of file
+}
```
Comment (alexwolfe) [inline dev/scss/_search-bar.scss]: Remove all these extra lines at the end of the file. 👾

**Review by alexwolfe — COMMENTED** [review]
@thaitwo ok, done reviewing, just a couple of comments. 👌  

Ping me when you make the updates and I'll review and merge. nice work.

**Comment by thaitwo** [comment]
@alexwolfe Thanks for the review! The updates are all set.

**Comment by alexwolfe** [comment]
Merged! 👍

**Comment by thaitwo** [comment]
Sweeeet! Thanks!! 😀

**Pull request merged**